src/main/java/Storage.java

```diff
@@ -0,0 +1,128 @@
+import java.io.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+
+/**
+ * Storage loads tasks from the file at <code>filePath</code> upon startup
+ * and saves tasks into the file at <code>filePath</code>
+ * every time there is changes to the to do list.
+ *
+ */
+public class Storage {
+    private File f;
+    /**
+     * Constructor for Storage class.
+     * @param filePath relative directory of the file.
+     */
+    public Storage(String filePath) {
+        this.f = new File(filePath);
+    }
+
+    /**
+     * Load the data from the saved file into the current to do list. load() is called upon startup.
+     * The file is read as <code>String</code> and being translated into a <code>Task</code>.
+     * @return ArrayList that contains all Task in the current saved file.
+     * @throws DukeException Exception is thrown when there is no file with the <code>filePath</code>.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        if ((!f.exists())) {
+            throw new DukeException("☹ OOPS!!! Error: There is no previous file stored! ☹ OOPS!!!");
+        } else {
+            ArrayList<String> stringTasks = new ArrayList<String>();
+            ArrayList<Task> tasks = new ArrayList<Task>();
+            try {
+                BufferedReader br = new BufferedReader(new FileReader(f));
+                String text;
+                while ((text = br.readLine()) != null) {
+                    stringTasks.add(text);
+                }
+            } catch (IOException e) {
+                System.out.println("Error: no File detected. " + e.getMessage());
+            }
+
+            for (int i = 0; i < stringTasks.size(); i++) {
+                String task = stringTasks.get(i);
+                String type = task.charAt(1) + "";
+                switch (type) {
+                    case "T": {
```

Comment:
the letters `T`, `D`, etc. seem quite unmeaningful to me. Not sure if we can improve it to be clearer and easier to understand.